entity-framework/core/what-is-new/ef-core-5.0/whatsnew.md

```diff
@@ -2,17 +2,545 @@
+    }
+}
+```
+
+Notice that:
+* The interceptor has both sync and async methods. This can be useful if you need to perform async I/O, such as writing to an audit server.
+* The interceptor allows SaveChanges to be skipped using the `InterceptionResult` mechanism common to all interceptors.
+
+The downside of interceptors is that they must be registered on the DbContext when it is being constructed. For example:
+
+```c#
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        => optionsBuilder
+            .AddInterceptors(new MySaveChangesInterceptor())
+            .UseSqlite("Data Source = test.db");
+```
+
+In contrast, the events can be registered on the DbContext instance at any time.
+
+### Exclude tables from migrations
+
+It is sometimes useful to have a single entity type mapped in multiple DbContexts. This is especially true when uising [bounded contexts](https://www.martinfowler.com/bliki/BoundedContext.html), for which it is common to have a different DbContext type for each bounded context.
+
+For example, a `User` type may be needed by both an authorization context and a reporting context. If a change is made to the `User` type, then migrations for both DbContexts will attempt to update the database. To prevent this, the model for one Context can be configured to exclude the table from its migrations.
+
+In the code below, the `AuthorizationContext` will generate migrations for changes to the `Users` table, but the `ReportingContext` will not, preventing the migrations from clashing.
+
+```C#
+public class AuthorizationContext : DbContext
+{
+    public DbSet<User> Users { get; set; }
+}
+
+public class ReportingContext : DbContext
+{
+    public DbSet<User> Users { get; set; }
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<User>().ToTable("Users", t => t.ExcludeFromMigrations());
+    }
+}
+```
+
+### Required 1:1 dependents
+
+TODO
+
+### Options for migration generation
+
+EF Core 5.0 introduces greater control over generation of migrations for different purposes. This includes the ability to:
+
+* Know whether the migration is being generated for a script or for immediate execution
+* Know if an idempotent script is being generated
+* Know if the script should exclude transaction statements (See _Migrations scripts with transactions_ below.)
+
+This behavior is specified by an the `MigrationsSqlGenerationOptions` enum, which can now be passed to `IMigrator.GenerateScript`.
+
+Also included in this work is better generation of idempotent scripts with called to `EXEC` on SQL Server when needed.
+
+### Migrations scripts with transactions
+
+SQL scripts generated from migrations now contain statements to begin and commit transactions as appropriate for the migration. For example, the migration script below was generated from two migrations. Notice that each migration is now applied inside a transaction.
```

Comment:
A small note on the possibility of disabling this?

Reply:
Added, although it it mentioned above.